src/manifolds/MetricManifold.jl

```diff
@@ -51,6 +51,34 @@ inner product ``g(X, X) > 0`` whenever ``X`` is not the zero vector.
 """
 abstract type RiemannianMetric <: AbstractMetric end
 
+@doc raw"""
+    change_metric(M::AbstractcManifold, G2::AbstractMetric, p, X)
+
+On the [`AbstractManifold`](@ref) `M` with implicitly given metric ``g_1``
+and a second [`AbstractMetric`](@ref) ``g_2`` this method performs a change of metric in the
+sense that it returns the tangent vector `Z` for which
+
+```
+g_2(X,Y) = g_1(Z,Y) \quad \text{for all} Y ∈ T_p\mathcal M.
```

Comment:
Could you explain what this condition is useful for? I would've expected `g_2(X,Y) = g_1(change_metric(M, G2, p, X),change_metric(M, G2, p, Y))` to be more important?

Reply:
But that would mean you deine the change metric by using change metric? that is kind of recursive?

This equation is basically what you mentioned with the identity push forward, i.e. X and Z and ”the same“ just with respect to different metrics? Also I think in your equation the two change metrics within the G1 would either cancel (G2*G2^{-1}) or double (G2^2) so this does not seem to be right, I am not sure 100% (its late).

Reply:
No, that's not recursive -- it's the same thing I described in https://github.com/JuliaManifolds/Manifolds.jl/issues/17#issuecomment-915440745 . Here `change_metric` is the function `f` from my previous comment. You get an equation for matrix `B` which defines the transformation.

Reply:
Or in other words: What is your result, X is the input and it should hold for all Y, then your proposal has no Z to be computed/returned?

Reply:
Maybe my idea is incorrect but still I think it would be nice to explain what exactly do we need `change_metric` for.

Reply:
> No, that's not recursive -- it's the same thing I described in [#17 (comment)](https://github.com/JuliaManifolds/Manifolds.jl/issues/17#issuecomment-915440745) . Here `change_metric` is the function `f` from my previous comment. You get an equation for matrix `B` which defines the transformation.

my transformation matrix with this is B= G_1^-1G_2

Reply:
> Maybe my idea is incorrect but still I think it would be nice to explain what exactly do we need `change_metric` for.

If we just take R2, then a metric can be described by a SPD matrix. Let G1 = I be the default metric (that is g_1(x,y) = xTy) and G2 = 2I (and I think I even got the order right).

Assume you have a G2 unit vector with a certain direction (the direction will not change here because both are constant diagonal matrices). In order for it to be g1-unit norm (in general also rotation etc is adopted) you have to G_1^-1G2 the vector (or in other words multiply it by two :D )

Reply:
> > No, that's not recursive -- it's the same thing I described in [#17 (comment)](https://github.com/JuliaManifolds/Manifolds.jl/issues/17#issuecomment-915440745) . Here `change_metric` is the function `f` from my previous comment. You get an equation for matrix `B` which defines the transformation.
> 
> my transformation matrix with this is B= G_1^-1G_2

Well, it's not. For example when `g_N` (from my comment) is the identity, then `B` is square root of `g_M`.

> If we just take R2, then a metric can be described by a SPD matrix. Let G1 = I be the default metric (that is g_1(x,y) = xTy) and G2 = 2I (and I think I even got the order right).
> 
> Assume you have a G2 unit vector with a certain direction (the direction will not change here because both are constant diagonal matrices). In order for it to be g1-unit norm (in general also rotation etc is adopted) you have to G_1^-1G2 the vector (or in other words multiply it by two :D )

Maybe I don't understand something but...
```julia
julia> B = [2 0; 0 2]
2×2 Matrix{Int64}:
 2  0
 0  2

julia> X = [1, 0]
2-element Vector{Int64}:
 1
 0

julia> using LinearAlgebra

julia> X' * I * X
1

julia> Xb = B \ X
2-element Vector{Float64}:
 0.5
 0.0

julia> Xb' * B * Xb # norm of Xb in the new basis with your transform
0.5

julia> Xb = sqrt(B) \ X
2-element Vector{Float64}:
 0.7071067811865475
 0.0

julia> Xb' * B * Xb # norm of Xb in the new basis with my transform
0.9999999999999998
```

Reply:
Or I have a mistake in my derivation – then please correct it. I don‘t see where my mistake is, but I had hoped I got it correct, otherwise I would not have started this PR.

Reply:
Also what is “your” transformation? As in my first question above, your equation only contained X (the input) and Y (all Y) but no output Z?

Reply:
My transformation is `f: X -> B*X`, where `B` is a matrix such that `G_1 = B^T G_2 B`, where `G_1`  is the original metric and `G_2` is the new metric. Now `<f(X), f(Y)>_{G2} = (BX)^T G_2 (BY) = X^T B^T G_2 B Y = X^T G_1 Y = <X, Y>_{G1}`. Even in the case of norm, notice that the transformed vector is used in both arguments of the metric, not just one: `||X||_{G2} = \sqrt{<X,X>_{G2}}`.

Reply:
If `G_1=L_1 U_1` and `G_2=L_2 U_2` are Cholesky decompositions of metric tensors, then `B=U_2 \ U_1` should solve the equation I think.

Reply:
Ah I See my mistake at least; I still have to check the last step of yours (the one using LU). But I will then update the docs tomorrow.

Reply:
(it was that the equation has to hold for all Z, Y to determine B and then return the transformed X, i.e. BX).